### ISSUE TITLE
Serial gimbal does not work with Walksnail Gimbal firmware V3.2

### ISSUE THREAD

**Issue by mmosca** [iNavFlight/inav#10302]
## Current Behavior
INAV Serial Headtracker and/or Gimbal stop working after updating Gimbal FIrmware to V3.2 (or using new headtracker firmware on the goggles)

## Steps to Reproduce
1. Update Gimbal to V3.2
2. Gimbal stops responding to INAV position commmands via UART
or
1. Use release version of Goggles L firmware
2. INAV can't recognize head tracker input

## Expected behavior
1. Update Gimbal to latest version
2. Gimbal responds to INAV position commands via UART
or
1. Update Headtracker to latest version
2. INAV recognizes gimbal inputs

## Suggested solution(s)
We are working with Caddx to restore compatibility.

## Additional context
Feature was tested on gimbal firmware v3.0 and possibly an earlier version or the headtracker firmware.
Caddx is working on updated firmware that will restore compatibility.

**Comment by Jonbuckles** [comment]
Ive also tested this on FW3.3beta with no luck from https://www.allxianfei.com/en/h-pd-138.html
I noticed GimbalConfig software is a little different and has channel assignment dropdown menus. 
![image](https://github.com/user-attachments/assets/24c2379a-1067-404c-9e95-aae167c71fbd)

**Comment by Jonbuckles** [comment]
So I was able to successfully use the channel assignment tabs in the FW3.3 gimbal software above. 
I assigned my CRSF channels (8,9,10,11,12) to the MODE SENS ROLL PITCH YAW drop down menus in the software. 
I then connected a wire from my CRSF TX/GND (so TX/GND goes to FC AND Gimbal), to PWM1 input and GND input on the gimbal port. 
So basically bypassing UART control via inav for now. Hopefully this is useful in some way.

**Comment by mmosca** [comment]
Next ht tracker version (v1.9) will restore functionality.

**Comment by dzaro-dev** [comment]
@mmosca  How can you connect the gimbal to inav via UART? Is there a way?

I know the stright-forward way is to connect directly to the TX pad on the CRSF receiver, but is there another way to connect via a free UART port on the FC and if so, what should be selected under Peripheral?

**Comment by mmosca** [comment]
Yes. The gimbal can be controlled by uart. Pick Serial Gimbal on the peripheral list.

![image](https://github.com/user-attachments/assets/f33e3b3f-9d4d-419c-b4e2-933451a6e21a)

You do need the latest version of the firmware on the gimbal and headtrackers for compatibility.

**Comment by Drasnore** [comment]
Hello, sorry if I'm not supposed to reply here, but I'm having issue with Inav not detecting headtracker inputs with Inav 8.0.0-RC3, from with Goggles X with the headtracker module fw 1.9.
UART 4 is selected as Serial Headtracker, headtracker type is serial in the configuration tab.
I'm seeing no input in the debug logging with debugger type set as headtracker.

Any help is appreciated, thank you.

**Comment by str8man** [comment]
> Hello, sorry if I'm not supposed to reply here, but I'm having issue with Inav not detecting headtracker inputs with Inav 8.0.0-RC3, from my Goggles X with the headtracker module fw 1.9. UART 4 is selected as Serial Headtracker, headtracker type is serial in the configuration tab and it is wired to my vtx correctly. I'm seeing no input in the debug logging with debugger type set as headtracker.
> 
> Any help is appreciated, thank you.

Hello,
It's reassuring to know I'm not alone with this problem.
https://github.com/iNavFlight/inav/pull/10109#issuecomment-2558201580

**Comment by wilco1967** [comment]
Hello,

I replaced my GM3 gimbal, after I broke the old one (stupid me dropping it.... not crash related).

Trying to get the new GM3 working and updating the firmware. 
It works fine with gimbal firmware 3.4, but anything newer, 3.6, or 3.8 , iNav gimbal control doesn't seem to work.

Back to 3.4, and it works fine again.

Is this a know issue, or is there some different setting I need to make ?

iNav version 8.01
